Port some of autests to OS X

gtenev opened the pull request:
* changed `netstat` usage to `lsof`
* added OS X specific `sysctl` usage for getting port range
* changed less portable `SOL_TCP` to `IPPROTO_TCP`

masaori335 commented:
[approve ci autest]

dragon512 commented:
I have some fixes coming in to help deal with double, txn failures.
are the tests you tweaked failing because the system does not have lsof on it?

gtenev commented:
Here is more info about the changes in case this is what you were asking about:
* tweaked few tests to use `lsof` as a more portable way to get the listening tcp ports instead of `netstat` (on OS X `netstat` is different and lacks the options used in the tests)
* changed the `sysctl` state variable names for determining ephemeral ports range on OS X since whatever was used is not available on OS X
* changed the less portable SOL_TCP to IPPROTO_TCP (which fails the building of some of the plugins generated during the AuTest bench test run)

If you were asking about why the Jenkins AuTest job failed then I was not able to figure it out just by looking at the job console output (at the time this PR was created I had them succeed **locally** on both OS X and CentOS 6)

I wonder if it would be a good idea to have the ability to trigger by hand a second job run with `--debug` or something of that sort after the initial AuTest job fails.

dragon512 commented:
That might be a good idea to add a secondary --debug --verbose run if we have failures... at least with the one that failed. I thing @zwoop is already complaining the time for the 60 or so test we have is taking to long ( ie ~10 minutes). I welcome the changes, as I am myself been making fixes to address testing issues on a larger test set of OS in our internal CI at work. I try to check this out tomorrow and rebase with the latest test fixes today and see what happens for me. I would love to fix the issue on the make, but honestly before this PR i never heard of lsof ( I did see the issue with netstat... ie why we get a message on the Mac about netstat not being used)

gtenev commented:
@dragon512, I was thinking more in terms of triggering the `--debug --verbose` AuTest job by hand (sometimes it would not be necessary) with `[approve ci autest]` or something of the sort.

`netstat` dates long back in time and on different "Unix-like" systems its parameters may differ a lot (one could compare the FreeBSD and Linux man pages to get a sense). 

`lsof` leverages an old "[treats everything as a file](https://en.wikipedia.org/wiki/Everything_is_a_file)" philosophy but appeared fairly recently (still more than 17+ years ago, I think) and seems to be more portable (ported to a big number of "Unix-like" systems and having generally the same parameters and functionality)

masaori335 commented:
[approve ci autest]
[approve ci CentOS]
[approve ci Ubuntu]

dragon512 commented:
@gtenev I am adding your mac fix for port selection. in a different PR. I have a different fix for the netstat issue that I am adding to the core Autest code, and a different PR to address the change of netstat in the tests

dragon512 commented:
@gtenev I think you will want to double check this PR, I should have most the fixes mac issues for you. None of the tests should need netstat as When.PortOpen() should be doing something similar now.

zwoop commented:
@gtenev Where are we with this? Is this still WIP? Seems it has merge conflicts now :-).

bryancall commented:
[approve ci debian]

zwoop commented:
Going to close this for now. @gtenev Please (re)open once you have something we can land please.
